### DISCUSSION TITLE
[BOX32] Battlefield 2 Multiplayer over bf2Hub

### DISCUSSION THREAD

**Discussion by Moxnatiy** [ptitSeb/box64#2128] [Show and tell]
MacBook Pro 14 (M1 Pro) 
Running Ubuntu ARM64 VM over VMWare Fusion  
Wine 9.20 + box64 (box32)

Excellent performance > 100FPS in VM, Asahi linux 200-400FPS 

It still has bugs like can't move player camera 360*, sound, random crashes, BUT IT'S WORKS ! Awesome !

[YouTube](https://youtu.be/W0pRtgp9pk0)

**Comment by ptitSeb** [comment]
You should try `BOX64_DYNAREC_X87DOUBLE=1` and also `BOX64_DYNAREC_FASTROUND=0` and/or `BOX64_DYNAREC_FASTNAN=0` for the 360 degree issue

**Reply by Moxnatiy** [reply]
I tried all 3 options, unfortunately it didn't help, now it looks like 270* already

**Reply by Moxnatiy** [reply]
I also noticed that if I change in game -> mouse sensitivity from 1.0 to 0.1 I get small area to control camera around 1-2* degrees  up, down, and left, right.

**Reply by Moxnatiy** [reply]
also I try a BOX64_DYNAREC=0, and get 1h loading and 0.2Fps and same issue. are it's wine issue ?

**Reply by ksco** [reply]
Not necessary, well, you might need to try the same setup without Box64 on a x86-64 machine to be sure..

**Reply by Moxnatiy** [reply]
Looks like i found something interesting. So here is [demo](https://youtu.be/-vxgRaaqjTo)
Monitor is 2560x1080


1. The rotation angle depends on the window resolution, so if we set bigger resolution I get more rotation angle. 
2. if I disable virtual desktop and **set game resolution to 1920x1080 issue was gone !** , but it's not works on other resolutions 
3. I accidentally pressed alt+tab and noticed that the camera stops moving exactly when the cursor goes outside the window, I think maybe in the game it should be  ( fixed ) centered?

**Reply by Moxnatiy** [reply]
> Not necessary, well, you might need to try the same setup without Box64 on a x86-64 machine to be sure..

Box64 + wine-wow64 works ok, but low performance 
fex + wine  works ok, ultra low performance 

also I check wine 9.0 and it's has same issue (current wine is 9.20)

**Comment by Moxnatiy** [comment]
Commit [2d4f073](https://github.com/ptitSeb/box64/commit/2d4f073925aebb93f9f19122fe5cd9c55caf1df3)
Max settings, Full screen, sound works (not tested on Asahi Linux)
Wine:  wine-9.22-staging-tkg-amd64 from [here](https://github.com/Kron4ek/Wine-Builds/releases/tag/9.22) 
![IMG_3735](https://github.com/user-attachments/assets/d84d248e-ddde-4dc0-a7bb-4019c311bc21)
